common/proto_utils.cpp

```diff
@@ -193,6 +196,16 @@ void MergeTransaction(Transaction& txn, const Transaction& other) {
       ->MergeFrom(other.internal().event_machines());
 }
 
+TxnReplicaId GetTransactionReplicaId(const Transaction* txn) {
+  // CHECK(txn->internal().type() == TransactionType::SINGLE_HOME ||
+  //     txn->internal().type() == TransactionType::LOCK_ONLY)
+  //     << "Only SINGLE_HOME and LOCK_ONLY transactions have a valid replica id";
```

Comment:
@ctring I included the glog header but CHECKs wouldn't compile, do you have to add the library to modules individually or something?
```
/usr/bin/ld: ../common/libcommon.a(proto_utils.cpp.o): in function `slog::GetTransactionReplicaId(slog::Transaction const*)':
/vagrant/common/proto_utils.cpp:203: undefined reference to `google::LogMessageFatal::LogMessageFatal(char const*, int)'
/usr/bin/ld: /vagrant/common/proto_utils.cpp:203: undefined reference to `google::LogMessage::stream()'
/usr/bin/ld: /vagrant/common/proto_utils.cpp:203: undefined reference to `google::LogMessageFatal::~LogMessageFatal()'
/usr/bin/ld: /vagrant/common/proto_utils.cpp:203: undefined reference to `google::LogMessageFatal::~LogMessageFatal()'
collect2: error: ld returned 1 exit status
```

Reply:
You can fix this by adding `common` to the `commands_test` test in CMakeLists.txt of `/test/`
```
target_link_libraries(commands_test common test_utils common)
```
I guess this is because we never specified the dependencies between `common` and the glog library directly but relied on the listed order of the libraries so here `common`, which indirectly linked to `commands_test` via `test_utils` was somehow listed after the glog library in the linking process.